web/react/components/posts_view_container.jsx

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+const PostsView = require('./posts_view.jsx');
+const ChannelStore = require('../stores/channel_store.jsx');
+const PostStore = require('../stores/post_store.jsx');
+const Constants = require('../utils/constants.jsx');
+const ActionTypes = Constants.ActionTypes;
+const Utils = require('../utils/utils.jsx');
+const Client = require('../utils/client.jsx');
+const AppDispatcher = require('../dispatcher/app_dispatcher.jsx');
+const AsyncClient = require('../utils/async_client.jsx');
+const LoadingScreen = require('./loading_screen.jsx');
+
+import {createChannelIntroMessage} from '../utils/channel_intro_mssages.jsx';
+
+export default class PostsViewContainer extends React.Component {
+    constructor() {
+        super();
+
+        this.onChannelChange = this.onChannelChange.bind(this);
+        this.onChannelLeave = this.onChannelLeave.bind(this);
+        this.onPostsChange = this.onPostsChange.bind(this);
+        this.handlePostsViewScroll = this.handlePostsViewScroll.bind(this);
+        this.loadMorePostsTop = this.loadMorePostsTop.bind(this);
+        this.postsLoaded = this.postsLoaded.bind(this);
+        this.postsLoadedFailure = this.postsLoadedFailure.bind(this);
+        this.handlePostsViewJumpRequest = this.handlePostsViewJumpRequest.bind(this);
+
+        const currentChannelId = ChannelStore.getCurrentId();
+        const state = {
+            scrollType: PostsView.SCROLL_TYPE_BOTTOM,
+            scrollPost: null,
+            numPostsToDisplay: Constants.POST_CHUNK_SIZE
+        };
+        if (currentChannelId) {
+            Object.assign(state, {
+                currentChannelIndex: 0,
+                channels: [currentChannelId],
+                postLists: [this.getChannelPosts(currentChannelId)]
+            });
+        } else {
+            Object.assign(state, {
+                currentChannelIndex: null,
+                channels: [],
+                postLists: []
+            });
+        }
+
+        this.state = state;
+    }
+    componentDidMount() {
+        ChannelStore.addChangeListener(this.onChannelChange);
+        ChannelStore.addLeaveListener(this.onChannelLeave);
+        PostStore.addChangeListener(this.onPostsChange);
+        PostStore.addPostsViewJumpListener(this.handlePostsViewJumpRequest);
```

Comment:
Because that's what stores are for. To store state. Both the client and server can chage the state of the stores. The views should be a reflection of the state of the stores. (we are not there but should be headed in that direction)